apps/core/src/utils/stake/getUnstakeDetailsFromEvents.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { UNSTAKING_REQUEST_EVENT } from '../../constants';
+import { UnstakeEventJson } from '../../interfaces';
+import type { IotaEvent } from '@iota/iota-sdk/client';
+
+export function getUnstakeDetailsFromEvents(events: IotaEvent[]): {
+    validatorAddress: string;
+    totalAmountWithoutRewards: bigint;
+    totalUnstakeAmount: bigint;
+    totalRewards: bigint;
+} {
+    const unstakeEvent = events.find(({ type }) => type === UNSTAKING_REQUEST_EVENT);
+    const eventJson = unstakeEvent?.parsedJson as UnstakeEventJson;
+
+    const totalAmountWithoutRewards = events?.reduce((sum, event) => {
+        return (
+            sum + Number((event.parsedJson as { principal_amount: number }).principal_amount || 0)
+        );
+    }, 0);
+
+    const totalRewards = events?.reduce((sum, event) => {
+        return sum + Number((event.parsedJson as { reward_amount: number }).reward_amount || 0);
+    }, 0);
+
+    const totalUnstakeAmount = BigInt(totalAmountWithoutRewards) + BigInt(totalRewards);
+
+    return {
+        validatorAddress: eventJson.validator_address || '',
+        totalAmountWithoutRewards: BigInt(totalAmountWithoutRewards),
+        totalRewards: BigInt(totalRewards),
+        totalUnstakeAmount,
```

Comment:
I think we can improve these names. `totalAmountWithoutRewards` is confusing to me as it seems like we are subtracting the rewards from the total value when we are not doing this. We get the amount that was staked, then the rewards obtained, and then we calculate the total by adding them together.

wdyt about -> unstakeAmount  / unstakeRewards / totalUnstakeAmount